modules/bpi/bpi.push.inc

```diff
@@ -164,6 +164,17 @@ function bpi_convert_to_bpi($node, $category, $audience, $with_images = FALSE, $
           $materials_drupal[] = $id;
         }
       }
+      elseif(isset($item['value'])) {
+        $endpoints = $item['value']->endpoints;
+        $data = array_filter($endpoints[LANGUAGE_NONE], function ($i) {
+          return $i['entity_type'] == 'ting_object';
+        });
+        $ting_data = current($data);
+        $ting_object = entity_load_single('ting_object', $ting_data['entity_id']);
```

Comment:
I would appreciate some defensive coding here.

- If the `$data` array is empty `$current` will return `false`.
- If the ting object entity with the provided id does not exist then `$ting_object` will be `false`.